warehouse/packaging/models.py

```diff
@@ -787,9 +801,12 @@ def uploaded_via_trusted_publisher(self) -> bool:
                     .as_string()
                     .is_not(None),
                 )
-            ).count()
-            > 0
-        )
+            )
+    
+    @property
+    def publisher_url(self) -> str:
+        events = self._get_trusted_publisher_events()
+        return events.first().additional["publisher_url"] if events.count() > 0 else None
```

Comment:
Note that `_get_trusted_publisher_events` might return events that don't have `Event.additonal["publisher_url"]`. This is for backwards compatibility before we had this field. I think in this case, it would be fine to skip verification.

I think there should be only one event like this per file, so this could be:
```suggestion
    @property
    def publisher_url(self) -> str:
        event = self._get_trusted_publisher_events().one_or_none()
        return event.additional["publisher_url"] if event else None
```